pages/api/roles/index.ts

```diff
@@ -68,12 +69,32 @@ function handler(req: NextApiRequest, res: NextApiResponse) {
   }
 }
 
-// TODO: Optional Search query. Will be done in another PR to unblock others.
-
 async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
+  const { companyId, searchQuery } = req.query;
+
+  const queryCompanyId = isInteger(companyId as string) ? Number(companyId) : undefined;
+
+  const searchWords = (searchQuery as string) ? (searchQuery as string).split(' ') : [];
```

Comment:
Do we know for sure that searchQuery is a string? I'm getting this error when I pass a number:
```error - TypeError: 1.split is not a function```

```suggestion
  const searchWords = typeof searchQuery === "string" ? (searchQuery as string).split(' ') : [];
```